graph/patterns/navigation-property.md

```diff
@@ -0,0 +1,174 @@
+ 
+Navigation properties defined within an entity are not returned by default when retreiving the representation of an entity unless explicity desired by a service.  
+
+Implementing support for accessing the "$ref" of a navigation property allows a caller to return just the URL of related resource. e.g. `/user/23/manager/$ref`. This is useful when a client wishes to identify the related resource but doesn't need all of its properties.
+
+## When to Use this Pattern
+------------------------
+
+### "has a" relationships
+
+The use of navigation properties is preferred over including an Id field to reference the related entity in a many-to-one relationship.  Id values require a client to make two round trips to retrieve the details of a related entity.  With a navigation property a client can retrieve a related entity in a single round trip. 
+
+Navigation properties are also useful when clients sometimes want to retrieve both the primary entity and the related entity in a single round trip.  The `expand` query parameter makes this possible.
+
+### "parent-child" relationships
+
+Resources that contain a parent Id property in a child resource can utilize a navigation property in the parent resource that is declared as a collection of child resources. If desirable, a parent navigation property can also be created in the child resource to the parent resource. This is usually not necessary as the parent URL is a subset of child resource URL. The main use of this would be when retrieving child resources and choosing to expand properties of the parent resource so that both can be retrieved in a single request.  
+
+`/invoice/{invoiceId}/items/{itemId}?expand=parentInvoice(select=invoiceDate,Customer)`
+
+### "child-parent" relationships
+
+One other use case is when child resources appear in a non-contained collection and there is a desire to access the canonical parent:
+
+`/me/pinnedChannels/{channelId}/team`
+
+## Example
+-------
+
+### Retrieving a related entity
+
+```http
+GET /users/{id}/manager
+
+200 OK
+Content-Type: application/json
+
+{
+  "@odata.type": "#microsoft.graph.user",
+  "id": "6b3ee805-c449-46a8-aac8-8ff9cff5d213",
+  "displayName": "Bob Boyce"
+}
+```
+
+This navigation property could be described with the following CSDL: 
+```xml
+<EntityType Name="user">
+  <NavigationProperty Name="manager" Type="user" ContainsTarget="false" >
+</EntityType>
+```
+
+### Retrieving a reference to a related entity
+
+```http
+GET /users/{id}/manager/$ref
+
+200 OK
+Content-Type: application/json
+
+{
+  "@odata.id": "https://graph.microsoft.com/v1.0/directoryObjects/6b3ee805-c449-46a8-aac8-8ff9cff5d213/Microsoft.DirectoryServices.User"
+}
+```
+Note: Currently the base URL returned in $ref results are incorrect.  In order to process these URLs the client will need to convert the URL to a Graph URL.
```

Comment:
It's really unfortunate that we have this bug in graph. I wonder if we could pick an example that wasn't broken.

Reply:
I documented it as a correct URL.  I'd rather have the set of examples be consistent around the manager navigation property.